design/secrets_provider_milestone1_design.md

```diff
@@ -0,0 +1,385 @@
+| 1    | Run as Job                  | Run as a K8s Job that deploys a Pod that terminates after task completion | - Minimal resources usage<br />- Minimal code changes<br />- Explicit user experience (user expects Job to run to completion quickly) | - "Job" is not an existing application identity granularity. <br /><br />- Deployment will need to be changed (from Job → Deployment) for Milestone 2 (support secrets rotation)<br /><br />- For **deployment**, **deploymentConfig** and **statefulSet** segregation of duties, this solution is not interchangeable with our other secrets retrieval solutions without policy changes. (e.g. secrets provider as init-container, conjur authn k8s secrets, any of our clients or Conjur REST API) |
+| 2    | Run as non-terminating  Pod | Run as a Pod that does not terminate automatically after task completion | - Simple to deploy<br />                                     | - Wastes resources because we aren't updating K8s secrets and up all the time<br /><br />- For **deployment**, **deploymentConfig** and **statefulSet** segregation of duties, this solution is not interchangeable with our other secrets retrieval solutions without policy changes. (e.g. secrets provider as init-container, conjur authn k8s secrets, any of our clients or Conjur REST API) |
+
+**Decision:** Run as a Job
+
+At current, a "Job" is not an existing application identity granularity. In terms of granularities, a "Job" is at the same level as "Deployment". From our understanding customers do not regularly use "Deployment" as their chose granularity and tend to use "Service Account" / "Namespace". Because of this, support for a "Job" granularity might not provide value when customers can still use "Service Account" / "Namespace".
+
+We decided to run as a Job instead of Deployment because for Milestone 1 we are concerned with delivering Conjur values only once. At this time we are not supporting updating K8s Secrets with Conjur secret across multiple runs so deploying a Pod that will run continuously in the environment of a customer is a waste of their resources. Especially because the process of retrieving secrets will only happen once, at the inital spin up of the Secrets Provider.
+
+Additionally, when using Deployment to spin up a pod and the task finishes, the will container exit and be deleted. Now that the Secrets Provider needs to sit separately, when that container exists so too will the pod. Therefore, when using Deployment, Kuberentes will try to restart the Pod. After a few attempted restarts, the pod will error in a BackOff state. Because of this, we are unable to use Deployment to spin up a pod and must use a Job.
+
+#### Flow
+
+The flow for Milestone 1 is similar to the user experience from Phase 1. For each step, we will go into more detail below.
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined 
+
+   - *Optional:* add label to K8s Secret ***(Milestone 1)***
```

Comment:
i think that this suggestion is unrelated to your design and adds noise. You can create a separate issue for the secrets-provider to add the ability to filter by a label and design it there. It has nothing to do with the effort to support multiple applications and i think your design will be easier to grasp without running in multiple directions.